_assets/patches/geth/0033-mailserver-response-contains-cursor.patch

```diff
@@ -0,0 +1,181 @@
+diff --git a/whisper/mailserver/mailserver.go b/whisper/mailserver/mailserver.go
+index 57e6505..77727ef 100644
+--- a/whisper/mailserver/mailserver.go
++++ b/whisper/mailserver/mailserver.go
+@@ -88,17 +88,22 @@ func (s *WMailServer) Close() {
+ 	}
+ }
+ 
+-func (s *WMailServer) Archive(env *whisper.Envelope) {
++func (s *WMailServer) Archive(env *whisper.Envelope) ([]byte, error) {
+ 	key := NewDbKey(env.Expiry-env.TTL, env.Hash())
+ 	rawEnvelope, err := rlp.EncodeToBytes(env)
+ 	if err != nil {
+-		log.Error(fmt.Sprintf("rlp.EncodeToBytes failed: %s", err))
+-	} else {
+-		err = s.db.Put(key.raw, rawEnvelope, nil)
+-		if err != nil {
+-			log.Error(fmt.Sprintf("Writing to DB failed: %s", err))
+-		}
++		err := fmt.Errorf("rlp.EncodeToBytes failed: %s", err)
```

Comment:
maybe we can add more context to the error message, e.g. `rlp.EncodeToBytes failed while archiving: %s`?